kong/api/arguments.lua

```diff
@@ -279,6 +281,63 @@ infer = function(args, schema)
 end
 
 
+local function decode_map_arg(name, value, container)
+  -- the meaning of square brackets varies depending on the method.
+  -- It is considered a map if the method is POST, PUT, or PATCH
+  -- otherwise square brackets are interpreted as LHS (EE only)
+  local method = get_method()
+  if method ~= "POST" and method ~= "PUT" and method ~= "PATCH" then
+    return nil, "maps not supported for this method"
+  end
```

Comment:
I suppose we don't need similar in `decode_array_arg` as it has no conflict with LHS?

Reply:
correct. The refactor uses a single function for decoding, but there's a similar logic that uses `is_map` to only run this check for maps, the reason is that the syntax of a map is identical to that of LHS, used for filtering on GET requests (in EE).